helm/values.yaml

```diff
@@ -0,0 +1,111 @@
+  interval: 15s
+  # ---------------------------------------------------------------------
+  # -- SQL exporter configuration
+  # -- For a comprehensive and comprehensively documented configuration
+  # --  file check out: https://github.com/burningalchemist/sql_exporter/blob/master/documentation/sql_exporter.yml
+  # ---------------------------------------------------------------------
+config:
+  global:
+    scrape_timeout: 10s
+    scrape_timeout_offset: 500ms
+    min_interval: 0s
+    max_connections: 3
+    max_idle_connections: 3
+  # ---------------------------------------------------------------------
+  # -- Target and collectors are not set so the chart is more flexible
+  # -- Please configure it yourself
+  # ---------------------------------------------------------------------
+  #  target:
+  #    data_source_name: 'sqlserver://prom_user:prom_password@dbserver1.example.com:1433'
+  #    collectors: [active_connections]
+  #  collectors:
+  #    - collector_name: active_connections
+  #      metrics:
+  #        - metric_name: active_connections
+  #          type: gauge
+  #          help: 'Active connections'
+  #          key_labels:
+  #            - "datname"
+  #            - "usename"
+  #            - "state"
+  #          values:
+  #            - "count"
+  #          query_ref: active_connections
+  #      queries:
+  #        - query_name: active_connections
+  #          query: |
+  #                  SELECT
+  #                    datname::text,
+  #                    usename::text,
+  #                    state::text,
+  #                    COUNT(state)::float AS count
+  #                  FROM pg_stat_activity
+  #                  GROUP BY datname, usename, state;
+  #  collector_files:
+  #    - "*.collector.yml"
+
+# ---------------------------------------------------------------------
+# -- Collector Files
+# ---------------------------------------------------------------------
+# -- Collector files are mounter to /etc/sql_exporter/collectors dir
+# ---------------------------------------------------------------------
+# collectorFiles:
+#   pricing_data_freshness.collector.yml:
+#     collector_name: pricing_data_freshness
+#     metrics:
+#       - metric_name: pricing_update_time
+#         type: gauge
+#         help: 'Time when prices for a market were last updated.'
+#         key_labels:
+#           # Populated from the `market` column of each row.
+#           - Market
+#         static_labels:
+#           # Arbitrary key/value pair
+#           portfolio: income
+#         values: [LastUpdateTime]
+#         query: |
+#           SELECT Market, max(UpdateTime) AS LastUpdateTime
+#           FROM MarketPrices
+#           GROUP BY Market
+# ---------------------------------------------------------------------
+# -- These values only used for tests, users shouldn't set those
+# ---------------------------------------------------------------------
+tests:
+  serviceMonitor:
+    enabled: false
```

Comment:
@allanger I'd say, we can safely drop this whole block just to avoid confusion for users (even though the comment says it all). So the defaults aren't set, but it's ok since the template should include respective resources only if `tests.serviceMonitor.enabled` is explicitly set. What do you think?

Reply:
If we remove it from values, I'll have to add more complicated logic to templates, cause without this object, tests will become nil, and chart will not be installable at all, because the test template will be throwing panics. I can get rid of it and update templates to avoid it though, and you'll be able to say what you like more then

Reply:
@allanger Ah indeed! I've just checked out the PR, and bumped into nil stuff. 👍 
I applied a small change to the conditional wrapper like below:
```
{{- if (((.Values.tests).serviceMonitor).enabled)}}
...
{{- end }}
```
, and it seems to work just fine. I've also added the same wrapper in the `test-connection.yaml` file, otherwise it always gets rendered.

Please, let me know what you think, but to me it should be enough without spending extra time. I'm ready to merge the PR. 😃

Reply:
This condition doesn't seem a good option to filter out the `test-connection` test. And also, I thought that it would make sense to have test-connection always rendered, cause it can be used by users to see if their installation is set up correctly.

Reply:
I've updated the serviceMonitor test

Reply:
Yeah, agreed. Semantically this condition doesn't make sense for `test-connection` pod.

My concern is how useful is it to make it always rendered. Here's my point:

We have several cases when this test isn't indicative:

- Multiple connections (jobs): since targets are always named;
- Single named target.

We receive HTTP 200 with scrape duration, scrape errors and target up metrics for named targets. These values are to be scraped with Prometheus/AlertManager to react accordingly.

Situations when `sql_exporter` returns HTTP 500 are mostly for a single anonymous target:

- no metric families (`No metrics gathered`) - we don't return any of the above standard metrics, so there's literally nothing to show;
- no metrics encoded (`No metrics encoded`) - internal error due to the incorrect processing;

At the same time, I expect lots of errors in the beginning. Typically (based on the issues data), users install the exporter first, then start iterating to configure it. Most of the time they call the metric endpoint directly to check the data results until they're satisfied.

Reply:
I guess you're right. But I would use another condition to exclude it, wdyt?

Reply:
Yeah, let's give it a shot. I don't have a strong opinion here, since I still learn (and am amazed at times) how people integrate the app in their workflows. 👍

Reply:
I've updated the condition, but I'm not so sure about naming anymore. Maybe you can come up with something better, or if it's fine with you, I guess the PR is done. The problem is that currently tests are not running because they are detecting changes to find out what to test, so I've triggered them by upgrading the Chart version here: https://github.com/allanger/sql_exporter/actions/runs/6448821034/job/17506501142

Reply:
I think it's all good. 🎉 We can handle the tests problem separately if needed. 👍